StarTrek13/Ships/factions.dm

```diff
@@ -204,4 +218,45 @@ var/list/global/faction_spawns = list()
 /datum/alliance/proc/removeMember(datum/faction/D)
 	if(D in member_factions)
 		member_factions -= D
-		D.broadcast("Your faction has been removed from [name]!")
\ No newline at end of file
+		D.broadcast("Your faction has been removed from [name]!")
+
+
+////objectives code\\\\
+//Shitcoded by yours truly
+
+/datum/factionobjective
+	var/iscurrent = FALSE
+	var/datum/faction/assigned_faction
+	var/description = "ERROR: THIS MESSAGE SHOULD NOT BE DISPLAYED." //The "informative" message of the objective
+
+/datum/factionobjective/New()
+	..()
+
+/datum/factionobjective/proc/setup()
+	assigned_faction.broadcast("<font color='red'><B>An error has occured with faction objectives, or a coder forgot to change something.</B></font>")
+	return
+
+/datum/factionobjective/proc/check_completion() //SHOULD be called every so often, to check for completion.
+	return
+
+//EXAMPLE
+/datum/factionobjective/destroy1
+	var/obj/structure/overmap/ship/objective
+
+/datum/factionobjective/destroy1/setup() //shitty, but it's just an example for now. This means ALL ships, including your own faction's, will be able to be picked. Hopefully this will be fixed later.
+	if(!global_ship_list)
+		qdel(src)
+		return
+	objective = pick(global_ship_list)
+	description = "<B>Locate the target vessel and destroy it. Your target: [objective.name]</B>"
+	assigned_faction.broadcast("Your faction has been assigned an objective; [description]")
+	check_completion()
+
+/datum/factionobjective/destroy1/check_completion()
+	while(objective)
+		sleep(10)
```

Comment:
Just use addtimer to call the proc again in a second then return, if the objective still exists, otherwise sleep() will cause whichever proc called it to hang

Reply:
addtimer(CALLBACK(src, .proc/check_completion), 10, TIMER_STOPPABLE)
Would this be correct? I've never used it before.